### PR TITLE
Rank Change Experiment

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request
Changes the backend of how paygrades are worked out on equipment presets so we no longer need to override it several times for different roles based on playtime.

Added sub-10 hour ranks for most USCM roles (all enlisted that didn't already have them, and most officers above 2nd LT)
Added 70+ hour ranks for most USCM enlisted and some officers.

# Explain why it's good for the game
The hope is that this will make things a little more dynamic, and also help address the clutter we have of certain paygrades (IE the Sergeant SL leading 9 lance corporals and a dozen or more PFCs)
Whilst true, in regards to Squad Rifleman, there can now be MORE lance corporals, the hope is this will be more scattered and accounted for by senior NCOs being slightly more common.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
code: Changed back-end for working out equipment preset paygrades so as to remove the manual overrides on several roles based on playtime perks.
add: Added lower ranks for all enlisted and some officer roles subject to having played for less than ten hours. This rank cannot be used after playing more than ten hours, and ignores preferences for playtime perks.
add: Added a higher rank achievable to most enlisted and some officers, requiring 175+ hours.
add: Changed base rank for SL, SG, Spec and FTL by one grade up.
add: Changed low playtime rank for MP and Nurse from Lance Corporal to Corporal.
/:cl:

### CODE REVIEW

File: code/modules/gear_presets/_select_equipment.dm
Comment (realforest2001) [inline code/modules/gear_presets/_select_equipment.dm:654]:
```suggestion
/datum/equipment_preset/proc/spawn_weapon(gunpath, ammopath, atom/M, sidearm = 0, ammo_amount = 12)
```

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by efzapa** [comment]
Gunnery Sergeant Military Warden my beloved

**Review by Drulikar — CHANGES_REQUESTED** [review]
Only thing that stands out to me is a bunch of `paygrades` having only a `JOB_PLAYTIME_TIER_1` whereas others start/only have a `JOB_PLAYTIME_TIER_0`

File: code/modules/gear_presets/_select_equipment.dm
Comment (realforest2001) [inline code/modules/gear_presets/_select_equipment.dm:19]:
```suggestion
```
Remove this before full merge.

File: code/modules/gear_presets/_select_equipment.dm
Comment (realforest2001) [inline code/modules/gear_presets/_select_equipment.dm:86]:
```suggestion
```
Remove this before full merge.

**Comment by zzzmike** [comment]
round 23169 
Screenshot 2024-07-21 153548

not sure what caused the ??? rank but figured I'd post just in case. I like the PR concept in any case. Wiki is gonna be quite out of date after merge hehe

**Comment by rythenx** [comment]
I'm just a random player but I don't really like the idea of people with more playtime in the same job being able to lawfully order around other people of the same job.  I think this makes sense only for absolute beginners in a job, ie. being able to recognize a new rifleman by seeing they are a private is good because it leads to people helping them out.

I feel like having certain jobs like CMO be able to be two different ranks is just going to make people more judgemental toward "new" CMOs who don't have the higher rank and be less willing to listen to them.  It just feels like it would encourage elitism to me.

Additionally, it seems like in its current state this also leads to weird situations where a CE could be a higher rank than an ASO despite the CE being subordinate to the ASO.  Similarly it can result in the ASO and XO being the same rank despite the ASO being subordinate to the XO.

**Comment by efzapa** [comment]
> I'm just a random player but I don't really like the idea of people with more playtime in the same job being able to lawfully order around other people of the same job.  I think this makes sense only for absolute beginners in a job, ie. being able to recognize a new rifleman by seeing they are a private is good because it leads to people helping them out.  The same benefit doesn't really exist IMO when you are wanting to identify people with 10+ hours compared to people with 70+ hours.
> 
> I feel like having certain jobs like CMO be able to be two different ranks is just going to make people more judgemental toward "new" CMOs who don't have the higher rank and be less willing to listen to them.  It just feels like it would encourage elitism to me.
> 
> Additionally, it seems like in its current state this also leads to weird situations where a CE could be a higher rank than an ASO despite the CE being subordinate to the ASO.  Similarly it can result in the ASO and XO being the same rank despite the ASO being subordinate to the XO.

CMP can now *technically* get DASO'd if they yell at the CMO which is kinda fucky

**Comment by realforest2001** [comment]
Y'all keep forgetting that rank is not the be-all end-all in the military. Chain of Command is still a thing.

**Comment by rythenx** [comment]
> Y'all keep forgetting that rank is not the be-all end-all in the military. Chain of Command is still a thing.

True enough.  I was overthinking it and I do like the variety in ranks after seeing it for longer.

File: code/modules/gear_presets/uscm_ship.dm
Comment (Blundir) [inline code/modules/gear_presets/uscm_ship.dm:312]: So, anything about it? Literally SL and Warden can be same rank of QM right now, not even taking into consideration that CE got even higher ranks with this pr, and QM not even getting closer the one got stripped by Morrow.

**Review by Drulikar — APPROVED** [review]

**Comment by wgtjunior743** [comment]
Besides the corporate liason, there is no JOB_PLAYTIME_TIER_4 new roles.
image

the FTL did not get a rank up, their previous base rank was corporal and they can still be a corporal as base, the only difference is that they can rank up to sgt like the SG and spec.
image

**Pull request merged**
